src/biotite/structure/charges.py

```diff
@@ -138,6 +137,78 @@
 # electronegativity values)
 EN_POS_HYDROGEN = 20.02
 
+
+def _determine_aromatic_nitrogen_hybridisation(
+    elements, types, bond_types
+):
+    """
+    Determine whether a nitrogen atom involved in an aromatic system
+    possesses exclusively single bonds or one double bond as well.
+
+    The need to determine this arises from the fact that the assignment
+    of parameters for electronegativity computation is based on the
+    hybridisation state, i. e. no special parameters for the involvement
+    in aromatic systems are available. This poses no problem in the case
+    of the elements oxygen and sulfur as they are exclusively involved
+    in aromatic systems having single bonds on either side, enabling the
+    the assignment of aromaticity to a distinct hybridisation state.
+    Nitrogen, however, can be involved in aromatic systems having both
+    exclusively single bonds as well as one double bond, making the
+    assignment of aromaticity to a distinct hybridisation state
+    impossible. Examples for nitrogen atoms involved in an aromatic
+    system having exclusively single bonds as well as one double bond
+    within one and the same molecule are the aromatic compounds
+    'imidazole' and 'pyrazole'.
+
+    Parameters
+    ----------
+    elements: ndarray, dtype=str
+        The array comprising the elements of the inserted `atom_array`.
+    types: ndarray, shape=(n,m), dtype=int
+        The array containing the bond types of each bond of every atom.
+    bond_types: ndarray, dtype=int
+        The array containing information about the highest bond type of
+        the respective atom (except for BondType.AROMATIC).
+
+    Returns
+    -------
+    bond_types: ndarray, dtype=int
+        The inserted array in which the BondType.AROMATIC is replaced by
+        the a BondType revealing the bond order in case of nitrogen
+        atoms.
+    """
+
+    nitrogen_indices = np.where(elements == "N")[0]
+    for i in nitrogen_indices:
+         if BondType.AROMATIC in types[i]:
+            considered_row = types[i]
+            considered_row.sort()
+            # Aromaticity implies molecular cyclicality, i. e.
+            # an atom involved in an aromatic system has at
+            # least two bonds with the aromatic bond type
+            # Nitrogen has at most three bonds if involved in an
+            # aromatic system, where the third bond type is
+            # `single`
+            # Therefore, the presence of a third bond type
+            # indicates a sp3 hybridisation, whereas the absence
+            # of a third bond type can be either due to sp2
+            # hybridisation or deprotonation
+            # In order to account for this ambiguity, the charge
+            # is considered in case that a third bond type is
+            # not present
+            try:
+                considered_row[-3]
```

Comment:
Do you check here, whether `consider_row` has length 3? In this case I think its clearer to use `if len(...) >= 3 ... else ...`

Reply:
You are right, I'll correct it.